docs/4.4/api-reference.md

```diff
@@ -176,9 +428,60 @@ if err := client.DeleteRole(ctx, "auditor"); err != nil {
 
 ## Tokens
 
-[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add leaf clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+Teleport is a "clustered" system, meaning it only allows access to hosts that had been previously granted cluster membership. To achieve this, a cluster has "join tokens" which can be shared to extend trust.
+
+A remote host can exchange one of these tokens with the cluster's auth server to receive signed certificates and become a trusted Teleport host (auth, node, or proxy server). Likewise, a Teleport cluster can exchange a token to become a [trusted cluster](admin-guide.md#trusted-clusters).
+
+These tokens can be predefined static tokens, or dynamic tokens with a short life time. The latter can be generated by `tctl` or this API, and is more secure.
+
+You may want to use this API to manage tokens if:
+
+- You have a program that dynamically adds new hosts to clusters
+- You want to programmatically add leaf clusters to a trusted cluster
 
-**Retrieve token**
+### The Token Object
+
+The Token has a Roles field, which defines what roles this token provides in the root cluster.
+
+`Role` is a custom string type with pre-defined values, e.g. `RoleAuth`, `RoleNode`, `RoleProxy`, and `RoleTrustedCluster`. You can check the teleport package to see more.
+
+```go
+type ProvisionTokenV2 struct {
+  // Model Fields are fields that all Teleport objects have - see above
+  Model Fields
+  // Spec is the token specification
+  Spec ProvisionTokenSpecV2 struct {
+    // Roles is a list of roles associated with the token, that will be converted to
+    // metadata in the SSH and X509 certificates issued to the user of the token
+    Roles []teleport.Role // string
+  }
+}
+```
+
+**Role**
+
+Unlike the RBAC roles in the documentation above, the `Roles` field on a Token determines what server roles a new host can take on in a cluster.
+
+These are all of the possible role values, which can be found in the `teleport` package.
+
+```go
+RoleAuth           Role = "Auth"
+RoleWeb            Role = "Web"
+RoleNode           Role = "Node"
+RoleProxy          Role = "Proxy"
+RoleAdmin          Role = "Admin"
+RoleProvisionToken Role = "ProvisionToken"
+RoleTrustedCluster Role = "Trusted_cluster"
+RoleSignup         Role = "Signup"
+RoleNop            Role = "Nop"
+RoleRemoteProxy    Role = "RemoteProxy"
+RoleKube           Role = "Kube"
+RoleApp            Role = "App"
+```
+
+### Retrieve Token
+
+This is the equivalent of `tctl get tokens/[tokenString]`.
```

Comment:
this `tctl` command doesn't work, even though `tokens` is the correct prefix. I tried `token` as well. Is there another command that achieves what `GetToken` does? If not I'll just remove this line completely.

Reply:
I think `tctl tokens ls` is the analog, although I don't know if `tctl tokens ls <tokenString>` works.